src/lib.rs

```diff
@@ -5,41 +5,38 @@
 mod constants;
 mod utils;
 mod pb;
+mod instruction;
 
 use pb::sf::solana::block_meta::v1::{
     Arg, BubblegumMeta, Collection, Creator, Message, MetadataArgs, Uses, Metadata, Output,
 };
-
 use substreams::log;
 use substreams_solana::pb::sf::solana::r#type::v1::Block;
-
-mod instruction;
 use instruction::{parse, Instruction};
 use utils::convert_to_date;
 use substreams::store::{StoreGet, StoreGetArray};
 
 #[substreams::handlers::map]
 fn map_block(block: Block, address_lookup_table_store: StoreGetArray<String>) -> Result<Output, substreams::errors::Error> {
-    // log::info!("{:#?}", block);
-
+    let slot = block.slot;
+    let parent_slot = block.parent_slot;
+    let timestamp = block.block_time.as_ref().unwrap().timestamp;
     let mut data: Vec<BubblegumMeta> = vec![];
 
-    for trx in block
-        .clone().transactions_owned()
-        .filter(|txn| txn.meta().is_some())
-    {
+    for trx in block.transactions_owned() {
```

Comment:
This is pretty much the biggest change that needed to be done. Some solana blocks are over 2mb so cloning the same block on every transaction on a block is costly. 

The ideal pattern is creating you needed variables at the top and using them as you go over transactions, compiled instructions and inner instructions.

Reply:
Got it. Will keep this in mind when raising the PR for the next Solana program.